File: public/benchmarks/simd/c/simd_scalar.c
#include <stdio.h>
#define N 20000000
double a[N];
int main() {
    for (int i = 0; i < N; i++) a[i] = (double)(i % 1000) * 0.001;
    double sum = 0.0;
    for (int i = 0; i < N; i++) sum += a[i] * a[i];
    printf("%.6f\n", sum);
    return 0;
}

File: public/benchmarks/simd/go/simd_scalar.go
package main

import "fmt"

const N = 20000000

var a [N]float64

func main() {
	for i := 0; i < N; i++ {
		a[i] = float64(i%1000) * 0.001
	}
	sum := 0.0
	for i := 0; i < N; i++ {
		sum += a[i] * a[i]
	}
	fmt.Printf("%.6f\n", sum)
}
